apps/web/src/components/tx/ConfirmTxDetails/index.tsx

```diff
@@ -3,38 +3,79 @@ import { Checkbox, Divider, FormControlLabel, Grid2 as Grid, Stack, StepIcon, Ty
 import commonCss from '@/components/tx-flow/common/styles.module.css'
 import { TxDetails } from './TxDetails'
 import ExternalLink from '@/components/common/ExternalLink'
-import { useContext, useState } from 'react'
+import { useCallback, useContext, useState } from 'react'
 import { SafeTxContext } from '@/components/tx-flow/SafeTxProvider'
 import SignOrExecuteFormV2 from '../SignOrExecuteForm/SignOrExecuteFormV2'
 import type { SignOrExecuteProps } from '../SignOrExecuteForm/SignOrExecuteFormV2'
 import useTxPreview from '../confirmation-views/useTxPreview'
+import Track from '@/components/common/Track'
+import { MODALS_EVENTS } from '@/services/analytics'
+import useWallet from '@/hooks/wallets/useWallet'
+import { isHardwareWallet, isLedgerLive } from '@/utils/wallets'
+
+const InfoSteps = [
+  {
+    label: 'Review what you will sign',
+    description: (
+      <Typography>
+        Signing is an irreversible action so make sure you know what you are signing.{' '}
+        <Track {...MODALS_EVENTS.SIGNING_ARTICLE}>
+          <ExternalLink href="https://help.safe.global/en/articles/276343-how-to-perform-basic-transactions-checks-on-safe-wallet">
+            Read more
+          </ExternalLink>
+        </Track>
+        .
+      </Typography>
+    ),
+  },
+  {
+    label: 'Compare with your wallet',
+    description: (
+      <Typography>
+        Once you click <b>Sign</b>, the transaction will appear in your signing wallet. Make sure that all the details
+        match.
+      </Typography>
+    ),
+  },
+  {
+    label: 'Verify with external tools',
+    description: (
+      <Typography>
+        You can additionally cross-verify your transaction data in a third-party tool like{' '}
+        <Track {...MODALS_EVENTS.OPEN_SAFE_UTILS}>
+          <ExternalLink href="https://safeutils.openzeppelin.com/">Safe Utils</ExternalLink>
+        </Track>
+        .
+      </Typography>
+    ),
+  },
+]
+
+const HardwareWalletStep = [
+  InfoSteps[1],
+  {
+    label: 'For hardware wallets',
+    description: (
+      <Typography>
+        If you&apos;re using a hardware wallet with "blind" signing, please compare what you see on your device with the
+        hashes on the right.
+      </Typography>
+    ),
+  },
+  InfoSteps[2],
+]
 
 export const ConfirmTxDetails = (props: SignOrExecuteProps) => {
   const { safeTx, txOrigin } = useContext(SafeTxContext)
   const [txPreview] = useTxPreview(safeTx?.data)
   const [checked, setChecked] = useState(false)
+  const wallet = useWallet()
+  const showHashes = wallet ? isHardwareWallet(wallet) || isLedgerLive(wallet) : false
```

Comment:
Without the `SafeTx`, it's not possible to compute the necessary hashes, making external verification impossible.

A couple instances that come to mind:
1. On-chain confirmations: The `safeTxHash` (which cannot be computed) is signed.
2. Ledger Live: The interface displays the hashes, but the device shows the `SafeTx`.

Reply:
Not sure I follow, how can there be no SafeTx?

Reply:
Apologies, I had my reasoning reversed.

Without the hashes, external tools cannot verify them, and the `safeTxHash` is approved in parent Safes.